src/codec/cbor.rs

```diff
@@ -121,7 +121,7 @@ where
         // If we ran out before parsing, return none and try again later
         let res = match res {
             Ok(v) => Ok(Some(v)),
-            Err(e) if e.is_eof() => Ok(None),
+            Err(e) if e.is_eof() => return Ok(None),
             Err(e) => Err(e.into()),
```

Comment:
In case of an `Err` I don't see any value in advancing the buffer. I can see two scenarios on an error:

1. A user handles the error and drops the `Framed`.
2. A user handles the error and extracts the underlying I/O resource plus the buffer from the `Framed` via `Framed::into_parts`.

In the case of (1) there is no value in advancing the buffer on value. In the case of (2) advancing the buffer might actually be problematic, e.g. maybe the user wants to try again with a different codec.

With the above in mind, what do you think of the change below. Namely what do you think of not advancing the buffer on error?

``` diff
modified   src/codec/cbor.rs
@@ -119,10 +119,10 @@ where
         let res: Result<Dec, _> = serde::de::Deserialize::deserialize(&mut de);
 
         // If we ran out before parsing, return none and try again later
-        let res = match res {
-            Ok(v) => Ok(Some(v)),
-            Err(e) if e.is_eof() => Ok(None),
-            Err(e) => Err(e.into()),
+        let item = match res {
+            Ok(item) => item,
+            Err(e) if e.is_eof() => return Ok(None),
+            Err(e) => return Err(e.into()),
         };
 
         // Update offset from iterator
@@ -131,7 +131,7 @@ where
         // Advance buffer
         buf.advance(offset);
 
-        res
+        Ok(Some(item))
     }
 }
```

Reply:
Yeh, that all makes sense, I'll add this change.